source/lmp/pair_deepmd.cpp

```diff
@@ -1279,6 +1292,9 @@ void PairDeepMD::init_style() {
   if (out_each == 1) {
     int ntotal = atom->natoms;
     int nprocs = comm->nprocs;
+    if (ntotal > max_nlocal) {
```

Comment:
please make sure here the `atom->natoms` gives the total number of atoms in the system, not the natoms per processor.

Reply:
https://docs.lammps.org/Developer_atom.html#owned-and-ghost-atoms

Reply:
Maybe we should deprecate `all_nlocal` as LAMMPS has computed it for us per documentation.

Reply:
> please make sure here the `atom->natoms` gives the total number of atoms in the system, not the natoms per processor.

As tested and verified, `atom->natoms` is always the **true** number of systems, regardless of MPI or serius execution. While `all_local` might be 0 in all but one processes, which might make `stdf_comm_buff_size` to be kept on the original `atom->natoms`. To safely make sure no invalid read or write, is it safer to use `atom->natoms` instead of `all_natoms`?

Reply:
You are right. It's not safe when using MPI.

Let's directly define `int all_nlocal = atom->natoms`.

https://github.com/deepmodeling/deepmd-kit/blob/e6acacfa5c7dc6a2528d6828ecec187bb85c3806/source/lmp/pair_deepmd.cpp#L724-L725